kola/harness.go

```diff
@@ -348,9 +348,10 @@ func runTest(h *harness.H, t *register.Test, pltfrm string) {
 
 	// Cluster -> TestCluster
 	tcluster := cluster.TestCluster{
-		H:           h,
-		Cluster:     c,
-		NativeFuncs: names,
+		H:             h,
+		Cluster:       c,
+		NativeFuncs:   names,
+		TorcxManifest: torcxManifest,
```

Comment:
Do we want to pass the entire torcx manifest to the tests and leave them to have to deal with spinning up the machines or would it make more sense to make copies of the tests here where we specify the URL/version to use for each torcx package of the thing being tested.

Something like building a `TestedTorcxBinary` flag on the test which would declare say `docker` as the binary being tested, then we do the logic here to span out the versions and automatically insert them into the userdata as well as setting a variable into the `TestCluster` which says the version being downloaded.

This would allow us to run each one as a subtest for results/reporting purposes as well as allowing future filtering done from the cmdline argument level rather than being inserted into individual tests.

Looking at your test in the other commit, we could still pass the entire `TorcxManifest` as well if we wanted to have a test(s) specifically just to validate that certain versions are present for each binary.

Reply:
I think having some specialized utility code to do the "torcx package placing and looping" logic would be good if we did this sort of test a lot.

For now, this is the only test that needs it, so I'd rather not do it until we're sure we need it.

This already uses subtests for reporting:

```
=== RUN   docker.torcx-manifest-pkgs           
=== RUN   docker.torcx-manifest-pkgs/torcx-pkg-17.06
=== RUN   docker.torcx-manifest-pkgs/torcx-pkg-17.06/docker-info
=== RUN   docker.torcx-manifest-pkgs/torcx-pkg-17.06/resources
=== RUN   docker.torcx-manifest-pkgs/torcx-pkg-17.06/networks-reliably
=== RUN   docker.torcx-manifest-pkgs/torcx-pkg-17.06/user-no-caps
=== RUN   docker.torcx-manifest-pkgs/torcx-pkg-1.12
=== RUN   docker.torcx-manifest-pkgs/torcx-pkg-1.12/docker-info
=== RUN   docker.torcx-manifest-pkgs/torcx-pkg-1.12/resources
=== RUN   docker.torcx-manifest-pkgs/torcx-pkg-1.12/networks-reliably
=== RUN   docker.torcx-manifest-pkgs/torcx-pkg-1.12/user-no-caps
--- PASS: docker.torcx-manifest-pkgs (461.45s)
    --- PASS: docker.torcx-manifest-pkgs/torcx-pkg-17.06 (155.67s)
        --- PASS: docker.torcx-manifest-pkgs/torcx-pkg-17.06/docker-info (1.23s)
        --- PASS: docker.torcx-manifest-pkgs/torcx-pkg-17.06/resources (8.30s)
        --- PASS: docker.torcx-manifest-pkgs/torcx-pkg-17.06/networks-reliably (143.60s)
        --- PASS: docker.torcx-manifest-pkgs/torcx-pkg-17.06/user-no-caps (2.55s)
    --- PASS: docker.torcx-manifest-pkgs/torcx-pkg-1.12 (150.81s)
        --- PASS: docker.torcx-manifest-pkgs/torcx-pkg-1.12/docker-info (1.20s)
        --- PASS: docker.torcx-manifest-pkgs/torcx-pkg-1.12/resources (6.58s)
        --- PASS: docker.torcx-manifest-pkgs/torcx-pkg-1.12/networks-reliably (141.02s)
        --- PASS: docker.torcx-manifest-pkgs/torcx-pkg-1.12/user-no-caps (2.00s)
```

For now I don't think there's a compelling reason to over-abstract this when this is the only user of the torcx stuff. Longer term, maybe.